### PR TITLE
Filter select multiple genres

### PR DESCRIPTION
**Pull request by vanessaip**
https://alanyan.atlassian.net/browse/SG-103?atlOrigin=eyJpIjoiZWU3NWYxMDk5MTIzNDUxY2EzMzQyMDM1ZjhkMDhhOTUiLCJwIjoiaiJ9

### CODE REVIEW

File: frontend/screens/FilterScreen.tsx [hunk 1]
```diff
@@ -16,8 +16,8 @@ function FilterScreen({ navigation }) {
 
   const [text, onChangeText] = React.useState("");
   const [textLength, setTextLength] = React.useState(MAX_LENGTH);
-
-  const [genre, setGenre] = React.useState(-1);
+  
+  const [genres, setGenres] = React.useState(0);
```
Comment (vanessaip) [inline frontend/screens/FilterScreen.tsx:20]: genres is a bit mask that represents which genres are chosen (ie. 0 -> none, 1 -> first genre, 11 -> first and second genre)

File: frontend/screens/FilterScreen.tsx [hunk 2]
```diff
@@ -83,10 +83,10 @@ function FilterScreen({ navigation }) {
         </View>
         <Carousel
           title={"Genre"}
-          description={"Choose a mood from the ones below"}
+          description={"Choose a genre from the ones below"}
           choices={genreChoices}
-          selectedChoice={genre}
-          onChange={(choice) => { setGenre(choice) }}
+          selectedChoice={genres}
+          onChange={(choice) => { (choice < 0 || Object.is(choice, -0)) ? setGenres(genres & ~(1 << -choice)) : setGenres(genres | (1 << choice)) }}
```
Comment (vanessaip) [inline frontend/screens/FilterScreen.tsx:89]: choice will be negative or -0 if it was already selected and needs to be unselected.

File: frontend/components/filter/Carousel/index.tsx [hunk 3]
```diff
@@ -37,7 +37,23 @@ const Carousel: FunctionComponent<Props> = (props) => {
             <FlatList
                 style={styles.list}
                 data={props.choices}
-                renderItem={({ item, index }) => <ChoiceComponent choice={item} onPress={(event) => index === props.selectedChoice ? props.onChange(-1) : props.onChange(index)} selected={props.selectedChoice === index} />}
+                renderItem={({ item, index }) => <ChoiceComponent 
+                                                    choice={item} 
+                                                    onPress={
+                                                        (event) => { 
+                                                            if (item.isGenre && ((props.selectedChoice & (1 << index)) !== 0)) {
```
Comment (vanessaip) [inline frontend/components/filter/Carousel/index.tsx:44]: passes in negative index if a genre was already selected and needs to be unselected

**Comment by vanessaip** [comment]
Not sure if this was the right way to do it, let me know if there's a better way!

I only modified FilterScreen.tsx and Carousel/index.tsx, the rest I think were done automatically when I was setting up, I could remove them if they shouldn't be merged.

**Review by voctory — APPROVED** [review]
The bit mask implementation is super cool!!! You got me to look at bit operations in JS for the first time ever, I don't think I could've done it a better way... The other files that were changed shouldn't be a problem, works fine on my machine :)

**Pull request merged**